misc/semver-checker/modules/semver-checker-core/src/test/java/org/ballerinalang/semver/checker/ProjectUtil.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.semver.checker;
+
+import io.ballerina.projects.BuildOptions;
+import io.ballerina.projects.directory.BuildProject;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.StringJoiner;
+
+/**
+ * This is the class for getting the ballerina package instance from the text code.
+ */
+public class ProjectUtil {
+
+    public static final String BAL_TOML_FILE_NAME = "Ballerina.toml";
+    public static final String BAL_FILE_EXT = ".bal";
+    private static Path tempProjectDir;
+    private static final String TEMP_DIR_PREFIX = "semver-enforcing-dir-";
+    private static final String MAIN_FILE_PREFIX = "main-";
+    private static final String PACKAGE_ORG = "Semver_validator";
+    private static final String PACKAGE_NAME = "patch_validator";
+    private static final String PACKAGE_VERSION = "1.0.0";
+
+
+    public static BuildProject createProject(String mainBalContent) throws Exception {
+
+        try {
+            // Creates a new directory in the default temporary file directory.
+            tempProjectDir = Files.createTempDirectory(TEMP_DIR_PREFIX + System.currentTimeMillis());
+
+            tempProjectDir.toFile().deleteOnExit();
+
+            // Creates a main file and writes the generated code snippet.
+            createMainBalFile(mainBalContent);
+            // Creates the Ballerina.toml file and writes the package meta information.
+            createBallerinaToml();
+
+            BuildOptions buildOptions = BuildOptions.builder().setOffline(true).build();
+            return BuildProject.load(tempProjectDir, buildOptions);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
I think we can ignore this exception here since this is a test scenario. @NipunaRanasinghe  WDYT?

Reply:
I think we better throw a test exception in here so that the test will terminate. we dont need to print the stacktrace

Reply:
Noted